src/mongoc/mongoc-write-concern.c

```diff
@@ -205,6 +205,7 @@ mongoc_write_concern_set_wtimeout (mongoc_write_concern_t *write_concern,
                                    int32_t            wtimeout_msec)
 {
    bson_return_if_fail(write_concern);
+   bson_return_if_fail(wtimeout_msec >= 0);
```

Comment:
The MONGOC_WARNING() and friends are problematic for drivers built on top of mongoc.
It has no semantical understanding of what raised the warning or how to convey that to the caller.

If a function cal fail it must report that in form of a return value or out param. Or invent errno.

This specific case can be fixed by documenting it to be a programming error to pass wtimeout < 0, and say such value will be ignored.

The bson_return_if_fail() are for checking complex types/pointers, not to validate their value.
Illegal values should be checked for by normal input validation.